sys/net/gnrc/link_layer/lwmac/lwmac.c

```diff
@@ -0,0 +1,867 @@
+             * retransmit it (i.e., the retransmission scheme of LWMAC). */
+            if (gnrc_netdev->tx.packet != NULL) {
+                LOG(LOG_WARNING, "WARNING: [lwmac] TX %d times retry\n",
+                                 gnrc_netdev->tx.tx_retry_count);
+                gnrc_netdev->tx.state = TX_STATE_INIT;
+                gnrc_netdev->tx.wr_sent = 0;
+                lwmac_tx_update(gnrc_netdev);
+            }
+            else {
+                if ((pkt = gnrc_priority_pktqueue_pop(
+                         &gnrc_netdev->tx.current_neighbor->queue))) {
+                    gnrc_netdev->tx.tx_retry_count = 0;
+                    lwmac_tx_start(gnrc_netdev, pkt, gnrc_netdev->tx.current_neighbor);
+                    lwmac_tx_update(gnrc_netdev);
+                }
+                else {
+                    /* Shouldn't happen, but never observed this case */
+                    lwmac_set_state(gnrc_netdev, LWMAC_SLEEPING);
+                }
+            }
+            break;
+        }
+        case TX_STATE_FAILED: {
+            /* If transmission failure, do not try burst transmissions and quit other
+             * transmission attempts in this cycle for collision avoidance */
+            gnrc_netdev_lwmac_set_tx_continue(gnrc_netdev, false);
+            gnrc_netdev_lwmac_set_quit_tx(gnrc_netdev, true);
+            tx_success = "NOT ";
+            /* Intended fall-through, TX packet will therefore be dropped. No
+             * automatic resending here, we did our best.
+             */
+        }
+        case TX_STATE_SUCCESSFUL: {
+            if (gnrc_netdev->tx.current_neighbor == &(gnrc_netdev->tx.neighbors[0])) {
+                LOG(LOG_INFO, "[lwmac] Broadcast transmission done\n");
+            }
+            else {
+                LOG(LOG_INFO, "[lwmac] Transmission was %ssuccessful (%" PRIu32 " WRs sent)\n",
+                              tx_success, gnrc_netdev->tx.wr_sent);
+            }
+            lwmac_tx_stop(gnrc_netdev);
+
+            /* In case have pending packets for the same receiver, continue to
+             * send immediately, before the maximum transmit-limit */
+            if ((gnrc_netdev_lwmac_get_tx_continue(gnrc_netdev)) &&
+                (gnrc_netdev->tx.tx_burst_count < LWMAC_MAX_TX_BURST_PKT_NUM)) {
+                lwmac_schedule_update(gnrc_netdev);
+            }
+            else {
+                lwmac_set_state(gnrc_netdev, LWMAC_SLEEPING);
+            }
+            break;
+        }
+        default:
+            lwmac_tx_update(gnrc_netdev);
+    }
+
+    /* If state has changed, reschedule main state machine */
+    if (state_tx != gnrc_netdev->tx.state) {
+        lwmac_schedule_update(gnrc_netdev);
+    }
+}
+
+/* Main state machine. Call whenever something happens */
+bool lwmac_update(gnrc_netdev_t *gnrc_netdev)
+{
+    gnrc_netdev_lwmac_set_reschedule(gnrc_netdev, false);
+
+    switch (gnrc_netdev->lwmac.state) {
```

Comment:
In this function, there is no need to compare the changes of the related state (e.g., `gnrc_netdev->lwmac.state`), so we don't store the state variable here. 
While in functions like [`_rx_management`](https://github.com/RIOT-OS/RIOT/pull/6554/files#diff-889fe7d3645e899e8e09d05fa4996467R266), there is an [operation](https://github.com/RIOT-OS/RIOT/pull/6554/files#diff-889fe7d3645e899e8e09d05fa4996467R345) based on the changes of the state, so we store the state variable.